src/set_theory/cardinal.lean

```diff
@@ -929,6 +929,27 @@ by rw [← to_nat_cast 0, nat.cast_zero]
 lemma one_to_nat : cardinal.to_nat 1 = 1 :=
 by rw [← to_nat_cast 1, nat.cast_one]
 
+lemma to_nat_mul (x y : cardinal) : (x * y).to_nat = x.to_nat * y.to_nat :=
+begin
+  by_cases hx1 : x = 0,
+  { rw [comm_semiring.mul_comm, hx1, mul_zero, cardinal.zero_to_nat, nat.zero_mul] },
+  by_cases hy1 : y = 0,
+  { rw [hy1, cardinal.zero_to_nat, mul_zero, mul_zero, cardinal.zero_to_nat] },
+  cases lt_or_ge x cardinal.omega with hx2 hx2,
+  { cases lt_or_ge y cardinal.omega with hy2 hy2,
+    { rw [cardinal.to_nat_apply_of_lt_omega (cardinal.mul_lt_omega hx2 hy2),
+          cardinal.to_nat_apply_of_lt_omega hx2, cardinal.to_nat_apply_of_lt_omega hy2,
+          ←cardinal.nat_cast_inj, nat.cast_mul, ←classical.some_spec (cardinal.lt_omega.mp hx2),
+          ←classical.some_spec (cardinal.lt_omega.mp hy2),
+          ←classical.some_spec (cardinal.lt_omega.mp (cardinal.mul_lt_omega hx2 hy2))] },
+    { rw [cardinal.to_nat_apply_of_omega_le hy2, mul_zero, cardinal.to_nat_apply_of_omega_le],
+      rw [←not_lt, cardinal.mul_lt_omega_iff_of_ne_zero hx1 hy1, not_and_distrib, not_lt, not_lt],
+      exact or.inr hy2 } },
+  { rw [cardinal.to_nat_apply_of_omega_le hx2, nat.zero_mul, cardinal.to_nat_apply_of_omega_le],
+    rw [←not_lt, cardinal.mul_lt_omega_iff_of_ne_zero hx1 hy1, not_and_distrib, not_lt, not_lt],
+    exact or.inl hx2 },
+end
```

Comment:
This feels slightly cleaner:
```suggestion
begin
  by_cases hx1 : x = 0,
  { rw [comm_semiring.mul_comm, hx1, mul_zero, cardinal.zero_to_nat, nat.zero_mul] },
  by_cases hy1 : y = 0,
  { rw [hy1, cardinal.zero_to_nat, mul_zero, mul_zero, cardinal.zero_to_nat] },
  apply nat_cast_injective,
  rw nat.cast_mul,
  cases lt_or_ge x cardinal.omega with hx2 hx2,
  { cases lt_or_ge y cardinal.omega with hy2 hy2,
    { rw [cast_to_nat_of_lt_omega (cardinal.mul_lt_omega hx2 hy2),
          cast_to_nat_of_lt_omega hx2, cast_to_nat_of_lt_omega hy2] },
    { rw [cast_to_nat_of_omega_le hy2, mul_zero, cast_to_nat_of_omega_le],
      rw [←not_lt, cardinal.mul_lt_omega_iff_of_ne_zero hx1 hy1, not_and_distrib, not_lt, not_lt],
      exact or.inr hy2 } },
  { rw [cast_to_nat_of_omega_le hx2, _root_.zero_mul, cast_to_nat_of_omega_le],
    rw [←not_lt, cardinal.mul_lt_omega_iff_of_ne_zero hx1 hy1, not_and_distrib, not_lt, not_lt],
    exact or.inl hx2 },
end
```
which needs:
```lean
lemma nat_cast_injective : injective (coe : ℕ → cardinal) :=
to_nat_right_inverse.left_inverse.injective
```
```lean
@[simp]
lemma cast_to_nat_of_omega_le {c : cardinal} (h : omega ≤ c) :
  ↑c.to_nat = (0 : cardinal) :=
by rw [to_nat_apply_of_omega_le h, nat.cast_zero]
```
both of which should be easy to place